app/controllers/admin/tags_controller.rb

```diff
@@ -1,17 +1,23 @@
 class Admin::TagsController < Admin::BaseController
   before_action :find_tag, only: [:update, :destroy]
+  before_action :tags, only: [:index, :create]
 
   respond_to :html, :js
 
   def index
-    @tags = Tag.category.page(params[:page])
-    @tag  = Tag.category.new
+    @tag = Tag.category.new
   end
 
   def create
-    Tag.find_or_create_by!(name: tag_params["name"]).update!(kind: "category")
-
-    redirect_to admin_tags_path
+    @tag = Tag.category.new(tag_params)
+
+    if @tag.valid?
+      Tag.find_or_create_by!(name: @tag.name).update!(kind: "category")
+      
+      redirect_to admin_tags_path
+    else
+      render :index
+    end
```

Comment:
Here is just another way of doing this. I think it'd cover the same cases 🤔 .

```
  def create
    @tag = Tag.find_or_initialize_by(tag_params)
    if @tag.save && @tag.update!(kind: "category")
      redirect_to admin_tags_path
    else
      render :index
    end
  end
```

or 


```
  def create
    @tag = Tag.find_or_initialize_by(tag_params)
    if @tag.save
      @tag.update!(kind: "category")
      redirect_to admin_tags_path
    else
      render :index
    end
  end
```

Reply:
Very nice suggestion.